pintracker/optracker/operation.go

```diff
@@ -0,0 +1,192 @@
+package optracker
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	cid "github.com/ipfs/go-cid"
+
+	"github.com/ipfs/ipfs-cluster/api"
+)
+
+//go:generate stringer -type=OperationType
+
+// OperationType represents the kinds of operations that the PinTracker
+// performs and the operationTracker tracks the status of.
+type OperationType int
+
+const (
+	// OperationUnknown represents an unknown operation.
+	OperationUnknown OperationType = iota
+	// OperationPin represents a pin operation.
+	OperationPin
+	// OperationUnpin represents an unpin operation.
+	OperationUnpin
+	// OperationRemote represents an noop operation
+	OperationRemote
```

Comment:
It's not so much of a mixture. It fits pretty well the model. Effectively the OperationType+Phase types represents states. I don't see why this would need to be removed. Remote is simply an operation which requires no action itself, but overlapping Pin/Remote operations need to play by the same rules (cancelling each others etc) as overlapping Pin/Unpins.

Reply:
The way I see it, Pinning and Unpinning are operations, whereas Remoting isn't. Not a technical issue, just differs from how I think about the OperationTracker. Happy to leave as is.